omod/src/main/java/org/openmrs/web/controller/observation/ObsFormController.java

```diff
@@ -140,7 +141,12 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 						errors.reject("editReason", "Obs.edit.reason.empty");
 						return showForm(request, response, errors);
 					}
-					
+					if (obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
```

Comment:
@ibacher   hi ian,what do think of having this validation exception in legacy ui instead of core,i added a screen shot on the ticket Id,thanks cc @dkayiwa

Reply:
@HerbertYiga I think this check should be added to the [`ObsValidator`](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/validator/ObsValidator.java) rather than here. That way, it's also enforced inside the API.

Reply:
@ibacher   Thanks ian,let me do the fix there